src/process_utility.c

```diff
@@ -472,6 +474,225 @@ process_drop_trigger_start(ProcessUtilityArgs *args, DropStmt *stmt)
 	ts_cache_release(hcache);
 }
 
+static DDLResult
+process_drop_schema_start(DropStmt *stmt)
+{
+	/*
+	 * An error will be raised when we start dropping the functions used by a
+	 * background worker, so there is no point in doing anything here.
+	 */
+	if (stmt->behavior == DROP_RESTRICT)
+		return DDL_CONTINUE;
+
+	/*
+	 * Here we are relying on that if we fail to drop one of the
+	 * procedures/functions, this transaction will be rolled back so these
+	 * changes will not be committed.
+	 */
+	ScanIterator iterator =
+		ts_scan_iterator_create(BGW_JOB, RowExclusiveLock, CurrentMemoryContext);
+	ts_scanner_foreach(&iterator)
+	{
+		ListCell *cell;
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		bool schema_isnull, job_id_isnull;
+		int32 job_id = DatumGetInt32(slot_getattr(ti->slot, Anum_bgw_job_id, &job_id_isnull));
+		Name proc_schema =
+			DatumGetName(slot_getattr(ti->slot, Anum_bgw_job_proc_schema, &schema_isnull));
+		Ensure(!job_id_isnull, "corrupt job entry: job id is null");
+		Ensure(!schema_isnull, "corrupt job entry: schema for job %d is null", job_id);
+		foreach (cell, stmt->objects)
+		{
+#if PG15_GE
+			String *object = lfirst(cell);
```

Comment:
Well, for the >PG14 part yes, but for PG14 and less, not really because PG14 treats these values a little strange.

The contents of the node is marked as a `T_String` but the node should be of type `Value`. If you use this code:

```c
			Value *object = lfirst_node(Value, cell);
```

you will get an assertion failure.

If you try to use this code:

```c
			Value *object = lfirst_node(Node, cell);
```

You will get a compile error because there is no `T_Node` tag.

The best we can do is this code:

```c
#if PG15_GE
			String *object = lfirst_node(String, cell);
#else
			Value *object = lfirst(cell);
#endif
```